src/collective/cover/tiles/carousel.py

```diff
@@ -66,7 +65,10 @@ def populate_with_object(self, obj):
         :type uuids: List of strings
         """
         super(ListTile, self).populate_with_object(obj)  # check permission
-        uuids = ICoverUIDsProvider(obj).getUIDs()
+        if obj.portal_type == 'Collection':
```

Comment:
yes, I know; but I stumbled upon the folder use case in the process of fixing that and I came into the conclusion it doesn't make sense neither because of what I explained above. also, by removing that behavior we are able to remove those useless adapters, simplifying the code even more.